src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -738,10 +739,10 @@ func (b *ByocAws) Follow(ctx context.Context, req *defangv1.TailRequest) (client
 		var cancel context.CancelCauseFunc
 		ctx, cancel = context.WithCancelCause(ctx)
 		go func() {
-			if err := ecs.WaitForTask(ctx, taskArn, 3*time.Second); err != nil {
+			if err := ecs.WaitForTask(ctx, taskArn, 2*time.Second); err != nil {
 				if stopWhenCDTaskDone || errors.As(err, &ecs.TaskFailure{}) {
-					time.Sleep(time.Second) // make sure we got all the logs from the task before cancelling
-					cancel(err)
+					time.Sleep(2 * time.Second) // make sure we got all the logs from the task/ecs before cancelling
+					cancel(pkg.ErrDeploymentFailed{Service: "Defang CD", Message: err.Error()})
```

Comment:
This seems like the meat of the solution to the problem in the PR title, but I'm not sure I understand why we think this will fix it.

If I understand, we're decreasing the polling interval to every 2s while we wait for the cd task to be finish. Then, once it does, we're sleeping for an extra second longer and hoping that that's enough time to get the final logs entries from the stream.

I wonder if we should be sending some kind of "EOF" sentinel so we can have higher certainty that we're done.

Reply:
True. The "2 seconds" is based off the fact that CloudWatch Tail APIs seem to give us logs at 1 second bursts (in fact, the timestamps are rounded to 1s) and so the 2s is based on that, to ensure we get at least one 1s "heartbeat" before we terminate.

The CloudWatch Tail APIs do not send EOF even if the underlying task is done.